### DISCUSSION TITLE
Component will lose reactivity when toggling upon a props

### DISCUSSION THREAD

**Discussion by EiffelFly** [solidjs/solid#961] [Q&A] [answered]
Hi,

I just got in the vibe recently and the concept of reactivity is amazing, really appreciate it! In the meanwhile I got stuck in a problem that once I dynamic toggling the class upon a component's props, it will lost the reactivity. Would you mind shedding the light ahead.
 
```js
// this will work
<svg
  class={cn(
    'h-5 w-5 fill-sd-black',
    theme() === 'dark' ? 'hidden' : ''
  )}
  viewBox="0 0 16 16"
  xmlns="http://www.w3.org/2000/svg"
>
  <path d="<>" />
  <path d="<>" />
</svg>
```

```js
// this won't

<SunIcon
  styleName={cn(
    'w-5 h-5 fill-black',
    theme() === 'dark' ? '' : 'hidden'
  )}
/>
```

Here is the demo: [stackblitz](https://stackblitz.com/edit/solid-vite-idfic6)

**Comment by ryansolid** [comment, marked as answer]
Tracking reactivity is based on access. Destructuring/accessing props outside of primitives or the JSX loses reactivity. Inside SunIcon use `props` and then access the property like `props.styleName` where you use it.

**Reply by EiffelFly** [reply]
oh!!!! Got it, thanks for answering my question!
